src/language/ast.js

```diff
@@ -379,23 +396,26 @@ export type EnumValueNode = {|
   +value: string,
 |};
 
-export type ListValueNode = {|
+export type ConstListValueNode = ListValueNode<ConstValueNode>;
+export type ListValueNode<+V = ValueNode> = {|
```

Comment:
@leebyron Can we do it like this:
```js
type _ListValueNode<+V> = {|
  +kind: 'ListValue',
  +loc?: Location,
  +values: $ReadOnlyArray<ValueNode>,
  +values: $ReadOnlyArray<V>,
|};

export type ListValueNode = _ListValueNode<ValueNode>;
export type ConstListValueNode = _ListValueNode<ConstValueNode>;
```

That way it's more clear to the reader that we just reusing type definition.
Also, we not exposing the ability to change this template argument as part of public API.

Reply:
I tried that first actually, but we have a linter that blocks the use of `_` in type names.

Reply:
I instead followed the pattern I used for `ConstArgumentNode` and `ConstDirectiveNode` which I think is easier to follow and addresses your concern